lib/controllers/ephemeral/payment/payment_initiation_controller.dart

```diff
@@ -1,17 +1,27 @@
 import 'dart:async';
 import 'package:get/get.dart';
+import 'package:international_phone_input/international_phone_input.dart';
 import 'package:pispapp/models/phone_number.dart';
 
 class PaymentInitiationController extends GetxController {
   PaymentInitiationController();
 
   PhoneNumber phoneNumber;
 
+  RxBool validPhoneNumber = false.obs;
+
   void onPhoneNumberChange(PhoneNumber phoneNumber) {
     this.phoneNumber = phoneNumber;
+    _checkNumberValidity();
     update();
   }
 
+  Future<void> _checkNumberValidity() async {
+    PhoneService.parsePhoneNumber(phoneNumber.number, phoneNumber.countryCode).then((isValid) {
+      validPhoneNumber.value = isValid;
```

Comment:
are you going to prevent user to tap next if they havent provided a valid phone number?

Reply:
Yes, it is prevented via:
`Get.find<SetupController>().validPhoneNumber.value ?
                    () => Get.find<SetupController>().onPhoneNumberSubmitted() : null`
The onTap() is set to null if validPhoneNumber is false.

Reply:
Ohh okay, got it, thank you! Perhaps it would be better to change the color as well to inform people that they cannot go to the next screen instead of just setting the onTap to null

Reply:
Flutter buttons automatically change color when their onTap is not set so we get that behaviour for free

Reply:
Ohh I see cool, okay then!